### PR TITLE
jun-kazama's solution from AoC2017 day5

### PR DESCRIPTION
**Pull request by jun-kazama**
I have uploaded my solution from AoC2017 day5 to the fork without cloning it locally. This is my first time submitting a pull request, so any feedback or guidance would be greatly appreciated.

### CODE REVIEW

**Comment by ellnix** [comment]
Hi there Jun!

Thanks for making a pull request! Since this is your first time making one I'll go over some things, but don't be discouraged!

# Take care of your git history 

A pull request is a request from you to the repository owner (or maintainers) to pull your changes into the main repository. Many projects will merge your git history into the main project's git history too!

Currently, our repositories look like this:

jun kazama repo diagram before pull request

In many repositories, the maintainers will merge the changes like this:

jun kazama repository state after request is merged

The git history of a project is very important, as it keeps track of what was done, by whom, and when. Managing the git history is hard work sometimes, but if you do it right and you can make sure you present your changes as cleanly as possible. This makes it easier for anyone who is reviewing your changes in the pull request or far into the future when they're trying to find who made a particular change. 

# Work with git locally to understand it

In the future you will almost certainly be working with `git` from your command line. Even if you decide to use another tool, knowing how git works "under the hood" will make it much simpler to understand for you. For instance:
- `.gitkeep` files are only necessary if you are adding a folder that **has no files**. Git keeps track of **files**, so it does not include empty folders, as a workaround we add empty `.gitkeep` files inside of those empty folders.
- Most developers will try to include one change in one commit when they can.

# Try to fit in with the project

Very often a project that we are contributing to has some rules that we do not use ourselves (or don't agree with), but we comply with those rules in order to fit in. What you should check when you are first looking at contributing to a project:
- the `CONTRIBUTING` file (this repository doesn't have one) which has rules about how to format your code and your pull requests, and also what you should include
- any linter or formatter file (this repository doesn't have one)
- browse old pull requests and see how they were formatted, and if there was any discussion on them
- browse the existing code to see if someone has done something similar to you. If someone *has* done something similar you can try to base your work off of them to reduce the chance that you do something unusual that would cause the maintainers more work to fix.

For instance, you could check if someone else has submitted their solution to their repository. If you browse the code of this repository you could find [`2017/3/florence.go`](https://github.com/ellnix/aoc_club/blob/main/2017/3/florence.go). In that file you can see how someone else added a solution, they:
- went to the correct folder for the day: `2017/3/`
- added a file with their own name `florence.go`
- pasted their code into that file, along with some comments
- GitHub will tell you which commit last edited that file (this is called a `git blame`)
top bar git blame on github
commit history of one file on github
  - You can look at that commit and see how it was formatted, and base your work off that
  
Of course, you **don't have to take all these steps every time**! As you do this you will start to get a feeling for when you are confident you are contributing in the right way or not, and if you don't feel sure yet you can browse more code.

# In conclusion

These points are not very important for this one repository. I can fix all of them by accepting the pull request carefully. If you want to practice another pull request (or this one again) let me know. I'll hold off on merging for a couple of hours.
